### PR TITLE
partition_manager.py: Change span size calculation to include alignment

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Change the size of span partitions to include alignment paritions (EMPTY_x) so that the size reflects the size taken in memory, not just the sum of sizes of the included partitions.

Ref: NCSDK-16916

### CODE REVIEW

**Review by hakonfam — APPROVED** [review]

**Comment by joerchan** [comment]
Approving that the reported issue has been fixed:

Before:
```
+---0xe0000: nonsecure_storage (0x1a000 - 104kB)---+
| 0xe0000: settings_storage (0x2000 - 8kB)         |
| 0xe2000: EMPTY_1 (0x6000 - 24kB)                 |
| 0xe8000: pgps (0x18000 - 96kB)                   |
+--------------------------------------------------+
```
After:
```
+---0xe0000: nonsecure_storage (0x20000 - 128kB)---+
| 0xe0000: settings_storage (0x2000 - 8kB)         |
| 0xe2000: EMPTY_1 (0x6000 - 24kB)                 |
| 0xe8000: pgps (0x18000 - 96kB)                   |
+--------------------------------------------------+
```

@oyvindronningstad There is currently a number of workarounds in place for this issue, I could find 6 instances referencing this issue.
Should we remove these as part of this PR?

**Review by SeppoTakalo — APPROVED** [review]

File: scripts/partition_manager.py [hunk 1]
```diff
@@ -647,11 +647,13 @@ def set_size_addr(entry, size, address):
 
 def set_sub_partition_address_and_size(reqs, sub_partitions):
     for sp_name, sp_value in sub_partitions.items():
-        size = sum([reqs[part]['size'] for part in sp_value['span']])
+        end = max(((reqs[part]['address'] + reqs[part]['size']) for part in sp_value['span']))
+        address = min((reqs[part]['address'] for part in sp_value['span']))
+        size = end - address
+        # size = sum([reqs[part]['size'] for part in sp_value['span']])
```
Comment (sigvartmh) [inline scripts/partition_manager.py:653]: Why is this commented out part included?

Comment (oyvindronningstad) [reply scripts/partition_manager.py]: Thanks, removed

**Review by sigvartmh — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

**Review by VTPeltoketo — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -377,7 +377,9 @@ Scripts
 
 This section provides detailed lists of changes by :ref:`script <scripts>`.
 
-|no_changes_yet_note|
+* :ref:`partition_manager`:
+
+  * Change the size of span partitions to include alignment paritions (EMPTY_x) so that the size reflects the size taken in memory, not just the sum of sizes of the included partitions.
```
Comment (FrancescoSer) [inline doc/nrf/releases/release-notes-changelog.rst:382]:
```suggestion
  * The size of the span partitions was changed to include the alignment paritions (``EMPTY_x``).
  The size of the span partitions now reflects the memory space taken, not just the sum of the sizes of the included partitions.
```

**Review by FrancescoSer — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 2 above)
Comment (oyvindronningstad) [reply doc/nrf/releases/release-notes-changelog.rst]: Done, I also added a bit more clarifying information

**Review by FrancescoSer — APPROVED** [review]

**Comment by oyvindronningstad** [comment]
All failiing jobs have been rerun and are passing, so this is ready to merge.

**Pull request merged**